### ISSUE TITLE
Add support for lucid2

### ISSUE THREAD

**Issue by lsmor** [TristanCacqueray/haskell-xstatic#1]
Current `lucid-xstatic` depends on `lucid`. It would be nice to have a `lucid2-xstatic` package depending on `lucid2` (or maybe a `cabal` flag). The only difference with respect using `lucid` or `lucid2` is that `lucid2` doesn't have  `with` function, but it can be solved easily.

```haskell
-- | Adds 'script_' and 'link_' for javascript and css files.
xstaticScripts :: [XStaticFile] -> Html ()
xstaticScripts = traverse_ xrender
  where
    xrender :: XStaticFile -> Html ()
    xrender xf =
        let src = "/xstatic" <> decodeUtf8 (xfPath xf)
         in case xfType xf of
             -- "application/javascript" -> with (script_ mempty) [src_ src]               -- implementation with lucid
                "application/javascript" -> Lucid.script_ [Lucid.src_ src] ("" :: Html ())    -- implementation with lucid2
                "text/css" -> Lucid.link_ [Lucid.href_ src, Lucid.rel_ "stylesheet"]
                _ -> pure ()
```

**Comment by TristanCacqueray** [comment]
@lsmor thank you for the suggestion, I agree it would be nice, a new `lucid2-xstatic` package sounds good to me. Though I'm still blocked by https://github.com/haskell-servant/servant-lucid/issues/26

**Comment by TristanCacqueray** [comment]
Lucid2 is now supported with: https://hackage.haskell.org/package/lucid2-xstatic
